app/blog/article/[slug]/components/RelatedArticles.js

```diff
@@ -36,17 +34,21 @@ export default function RelatedArticles({ articles }) {
       </MarketingH2>
       {articles
         .slice(startIndex, startIndex + PAGE_SIZE)
-        .map((article, index) => (
-          <ArticleCard
-            key={index}
-            title={article.title}
-            summary={article.summary}
-            imageUrl={`https:${article.mainImage.url}`}
-            imageAlt={article.mainImage.alt}
-            linkUrl={`/blog/article/${article.slug}`}
-            showReadMoreButton
-          />
-        ))}
+        .map((article, index) => {
+          if (!article?.mainImage?.url) return null
```

Comment:
Isn't this going to leave gaps in the UI if the image is missing?

Seems like we need a `.filter(articleComponent => articleComponent && articleComponent !== null)` or something here.. no?